weasyl/test/login/test_authenticate_bcrypt.py

```diff
@@ -0,0 +1,126 @@
+# encoding: utf-8
+"""
+Test suite for: login.py::def authenticate_bcrypt(username, password, session=True):
+"""
+import json
+
+from weasyl.test import db_utils
+from weasyl import login, macro
+from weasyl import define as d
+
+
+# Main test password
+raw_password = "0123456789"
+
+
+class Bag(object):
+    def __init__(self, **kw):
+        for kv in kw.items():
+            setattr(self, *kv)
+
+
+def test_logins_fail_if_no_username_provided():
+    result = login.authenticate_bcrypt(username='', password='password')
+    assert result == (0, 'invalid')
+
+
+def test_logins_fail_if_no_password_provided():
+    user_name = "test"
+    result = login.authenticate_bcrypt(username=user_name, password='')
+    assert result == (0, 'invalid')
+
+
+def test_login_fails_if_incorrect_username_is_provided():
+    user_name = "test"
+    result = login.authenticate_bcrypt(username=user_name, password=raw_password)
+    assert result == (0, 'invalid')
+
+
+def test_login_fails_for_incorrect_credentials():
+    user_name = "testAuthBcry0001"
```

Comment:
Apparently `db` has session scope, so things aren’t cleaned up after each test as they should be. I would never have guessed given how slowly they run.